addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,194 @@
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpStatusCode;
+
+/**
+ * Active scan rule for Reverse proxy testing.
+ *
+ * @author Aman Rawat (@theamanrawat)
+ */
+
+public class ForbiddenBypassScanRule extends AbstractAppPlugin {
+
+    private static final String MESSAGE_PREFIX = "ascanalpha.forbiddenBypass.";
+    private static final Logger LOG = Logger.getLogger(ForbiddenBypassScanRule.class);
+
+    @Override
+    public int getId() {
+        return 40038;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public void scan() {
+        HttpMessage message = getNewMsg();
+        try {
+            sendAndReceive(message);
+            if (message.getResponseHeader().getStatusCode() == HttpStatusCode.FORBIDDEN) {
+                URI uri = message.getRequestHeader().getURI();
+                String host = uri.getAuthority();
+                String path = uri.getPath();
+                String schema = uri.getScheme();
+                sendPathPayloads(path, host, schema, uri);
+                sendHeaderPayloads(path, host, schema, uri); 
+            }
+        } catch (IOException e) {
+            LOG.error(e.getMessage(), e);
+        }
+    }
+    
+    public void sendPathPayloads(String path, String host, String schema, URI uri) throws IOException {
+        String[] pathPayloads =
+                {
+                        "/%2e" + path,
+                        path + "/./",
+                        "/." + path + "/./",
+                        path + "%20/",
+                        "/%20" + path + "%20/",
+                        path + "..;/",
+                        path + "?",
+                        path + "??",
+                        "/" + path + "//",
+                        path + "/",
+                        path + "/.testus",
+                        path + "../app.py"
+                };
+
+        for (String pathPayload : pathPayloads) {
+            String tmpUri = schema + "://" + host + "" + pathPayload;
+            HttpMessage reqWithPayload =
+                    new HttpMessage(new org.apache.commons.httpclient.URI(tmpUri, true));
+            sendAndReceive(reqWithPayload);
+
+            if (reqWithPayload.getResponseHeader().getStatusCode() == HttpStatusCode.OK) {
+                alertFound(uri, reqWithPayload);
```

Comment:
Should return on first alert found (and skip the header tests).

Reply:
It still needs to skip the header tests when an alert was already found.

Reply:
So How can I do that? I mean How can I know that alert was already found..

Reply:
Return a boolean (e.g. true alert found).

Reply:
Now have a look at updated file.